chef/data_bags/crowbar/migrate/provisioner/031_arch_repos.rb

```diff
@@ -0,0 +1,29 @@
+def upgrade ta, td, a, d
+  a['supported_oses'] = ta['supported_oses']
+
+  if a.fetch('suse', {}).fetch('autoyast',{})['repos']
+    a['suse']['autoyast']['repos'].keys.each do |os|
```

Comment:
Style/StringLiterals: Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping. (https://github.com/bbatsov/ruby-style-guide#consistent-string-literals)